newparameterizer/common/utils.go

```diff
@@ -0,0 +1,448 @@
+	}
+	name, ok := nameI.(string)
+	if !ok {
+		return "", fmt.Errorf("expected name to be of type string. Actual value %+v is of type %T", nameI, nameI)
+	}
+	return name, nil
+}
+
+// GetK8sResourcesWithPaths gets the k8s resources from a folder along
+// with the relaive paths where they were found.
+// Mutiple resources maybe specified in the same yaml file.
+func GetK8sResourcesWithPaths(k8sResourcesPath string) (map[string][]types.K8sResourceT, error) {
+	logrus.Trace("start GetK8sResourcesWithPaths")
+	defer logrus.Trace("end GetK8sResourcesWithPaths")
+	yamlPaths, err := common.GetFilesByExt(k8sResourcesPath, []string{".yaml"})
+	if err != nil {
+		return nil, err
+	}
+	k8sResources := map[string][]types.K8sResourceT{}
+	for _, yamlPath := range yamlPaths {
+		k8sYamlBytes, err := ioutil.ReadFile(yamlPath)
+		if err != nil {
+			logrus.Errorf("Failed to read the yaml file at path %s . Error: %q", yamlPath, err)
+			continue
+		}
+		currK8sResources, err := GetK8sResourcesFromYaml(string(k8sYamlBytes))
+		if err != nil {
+			logrus.Debugf("Failed to get k8s resources from the yaml file at path %s . Error: %q", yamlPath, err)
+			continue
+		}
+		relYamlPath, err := filepath.Rel(k8sResourcesPath, yamlPath)
+		if err != nil {
+			logrus.Errorf("failed to make the k8s yaml path %s relative to the source folder %s . Error: %q", yamlPath, k8sResourcesPath, err)
+			continue
+		}
+		k8sResources[relYamlPath] = append(k8sResources[relYamlPath], currK8sResources...)
+	}
+	return k8sResources, nil
+}
+
+// GetK8sResourcesFromYaml decodes k8s resources from yaml
+func GetK8sResourcesFromYaml(k8sYaml string) ([]types.K8sResourceT, error) {
+	// TODO: split yaml file into multiple resources
+
+	// NOTE: This roundabout method is required to avoid yaml.v3 unmarshalling timestamps into time.Time
+	var resourceI interface{}
+	if err := yaml.Unmarshal([]byte(k8sYaml), &resourceI); err != nil {
+		logrus.Errorf("Failed to unmarshal k8s yaml. Error: %q", err)
+		return nil, err
+	}
+	resourceJSONBytes, err := json.Marshal(resourceI)
+	if err != nil {
+		logrus.Errorf("Failed to marshal the k8s resource into json. K8s resource:\n+%v\nError: %q", resourceI, err)
+		return nil, err
+	}
+	var k8sResource types.K8sResourceT
+	err = json.Unmarshal(resourceJSONBytes, &k8sResource)
+	return []types.K8sResourceT{k8sResource}, err
+}
+
+// WriteResourceAppendToFile is like WriteResource but appends to the file
+func WriteResourceAppendToFile(k8sResource types.K8sResourceT, outputPath string) error {
+	logrus.Trace("start WriteResourceAppendToFile")
+	defer logrus.Trace("end WriteResourceAppendToFile")
+	yamlBytes, err := yaml.Marshal(k8sResource)
+	if err != nil {
+		logrus.Error("Error while Encoding object")
+		return err
+	}
+	// If the file doesn't exist, create it, or append to the file
+	f, err := os.OpenFile(outputPath, os.O_CREATE|os.O_APPEND|os.O_WRONLY, common.DefaultFilePermission)
```

Comment:
Do we want to maintain the same permissions and files as the original directory. For example, if a multiple artifacts are present in a single file, we could possibly retain the same structure for helm and possibly for kustomize too?